batcher/batcher.go

```diff
@@ -158,6 +161,17 @@ func (b *basicBatcher) Dispose() {
 	b.flush()
 	b.disposed = true
 	b.items = nil
+
+	// Drain the batch channel and all routines waiting to put on the channel
+DrainLoop:
+	for {
+		select {
+		case <-b.batchChan:
+		case <-time.After(5 * time.Millisecond):
```

Comment:
Seems like there could be a more intelligent/safer way to know if anyone is waiting to put stuff on the channel.